accord-core/src/main/java/accord/utils/async/AsyncResults.java

```diff
@@ -333,14 +333,19 @@ public RunnableResult(Callable<V> callable)
         @Override
         public void run()
         {
+            V result = null;
+            boolean success = false;
             try
             {
-                trySetResult(callable.call(), null);
+                result = callable.call();
+                success = true;
             }
             catch (Throwable t)
             {
                 trySetResult(null, t);
             }
+            if (success)
+                trySetResult(result, null);
```

Comment:
it would be good to document why the code is written this way, and we can avoid the boolean `success` if we rely on `return` when a failure is thrown

```
// There are two different type of exceptions: user function throws, listener throws.  To make sure this is clear,
            // make sure to catch the exception from the user function and set as failed, and let the listener exceptions bubble up.
            V call;
            try
            {
                call = callable.call();
            }
            catch (Throwable t)
            {
                setResult(null, t);
                return;
            }
            setResult(call, null);
```